build-tools/src/main/java/org/elasticsearch/gradle/LazyPropertyList.java

```diff
@@ -56,7 +56,7 @@ public Object[] toArray() {
 
     @Override
     public <T1> T1[] toArray(T1[] a) {
-        return delegate.stream().peek(this::validate).map(PropertyListEntry::getValue).collect(Collectors.toList()).toArray(a);
+        return delegate.stream().peek(this::validate).map(PropertyListEntry::getValue).toList().toArray(a);
```

Comment:
Can just be `toArray` directly

Reply:
We can't nicely do this here I think since we're creating a generic array type, it's just build code anyway :)

Reply:
Oh yes, didn't spot that